### PR TITLE
Speed-up of `phd` (through `hexdump_iter`)

### PR DESCRIPTION
**Pull request by br0ns**
Changed `util.fiddling.hexdump_iter` to take a file object instead of a string.  This massively speeds up `phd <file> | less` for huge files.

I also added a docstring to `util.fiddling.hexdump`.

### CODE REVIEW

**Comment by zachriggle** [comment]
:+1:

File: pwnlib/util/fiddling.py [hunk 1]
```diff
@@ -416,7 +416,7 @@ def rol(n, k, word_size = None):
 def ror(n, k, word_size = None):
     """A simple wrapper around :func:`rol`, which negates the values of `k`."""
 
-    return ror(n, -k, word_size)
+    return rol(n, -k, word_size)
```
Comment (zachriggle) [inline pwnlib/util/fiddling.py]: ????

Comment (br0ns) [inline pwnlib/util/fiddling.py]: Hm, I must have fixed that at some point and then forgotten about it.  It belongs in a separate PR but I'm too lazy :)
